runtime/onert/api/src/nnfw_api_internal.cc

```diff
@@ -1144,12 +1144,32 @@ NNFW_STATUS nnfw_session::train_prepare(const nnfw_train_info *info)
     return NNFW_STATUS_INVALID_STATE;
   }
 
-  // Use default info if info is null, otherwise use given info
-  (void)info;
+  nnfw_train_info tinfo;
+  if (info != nullptr)
+  {
+    tinfo = *info;
+  }
 
-  // NYI
-  // _state = State::PREPARED_TRAINING;
-  return NNFW_STATUS_ERROR;
+  onert::compiler::train::TrainingInfo training_info;
+  training_info.setBatchSize(tinfo.batch_size);
+  // TODO Set Loss function
+
+  try
+  {
+    auto compiler =
+      onert::compiler::CompilerFactory::get().create(_nnpkg, _coptions, &training_info);
```

Comment:
Now it's OK, but it's a little bit dangerous. 
We should confirm that `training_info` will not be used in `_compiler_artifact` (`_compiler_artifact->_executors`) because `training_info` will be free after this function is finished.

Reply:
How about passing by reference?

Reply:
How about passing `training_info` by const reference into `create()`?

Reply:
If you want to use `TrainingInfo` elsewhere other than `TrainingCompiler`, it's nice to have it in a session.
If not, it would be good for `TrainingCompiler` to have a copied `TrainingInfo` by passing it by const reference.